authz/rbac_translator_test.go

```diff
@@ -298,6 +651,90 @@ func TestTranslatePolicy(t *testing.T) {
 			}`,
 			wantErr: `"allow_rules" 0: "headers" 0: unsupported "key" :method`,
 		},
+		"bad audit condition": {
+			authzPolicy: `{
+				"name": "authz",
+				"allow_rules": [
+				{
+					"name": "allow_authenticated",
+					"source": {
+						"principals":["*", ""]
+					}
+				}],
+				"audit_logging_options": {
+					"audit_condition": "ABC",
+					"audit_loggers": [
+						{
+							"name": "stdout_logger",
+							"config": {},
+							"is_optional": false
+						}
+					]
+				}
+			}`,
+			wantErr: `Failed to parse AuditCondition ABC`,
+		},
+		"bad audit logger config": {
+			authzPolicy: `{
+				"name": "authz",
+				"allow_rules": [
+				{
+					"name": "allow_authenticated",
+					"source": {
+						"principals":["*", ""]
+					}
+				}],
+				"audit_logging_options": {
+					"audit_condition": "NONE",
+					"audit_loggers": [
+						{
+							"name": "stdout_logger",
+							"config": "abc",
+							"is_optional": false
```

Comment:
My understanding is if `is_optional` is true, this kind of configs will not cause failure but simply be ignored. The current impl will not work like this, right?

Reply:
I thought `is_optional` was about the actual logging/behavior level, so if it fails there it's okay? Please correct me if I'm wrong

I think at this level of code, the contents of `is_optional` aren't particularly relevant - it doesn't mean we are allowed to fail the parsing, right?

Reply:
We definitely should reject invalid Jsons. But specifically, the parsing fails here because we required `config` to be a struct (aka a Json object). If we allow it to be anything, then `"abc"` is a valid Json string.

By reading the current gRFC draft again, I realize I did state `config` should be an object. If it stays that way, yes we should be good here.

Reply:
IMO we should be good to force it to be a JSON object that is a map/dictionary/etc